src/commons/product_image_selection.rs

```diff
@@ -70,55 +73,67 @@ pub struct ResolvedProductImage {
 #[derive(Clone, Debug, Default, Deserialize, Eq, JsonSchema, PartialEq, Serialize)]
 #[serde(rename = "PascalCase")]
 #[derive(AsRefStr)]
+/// We default to `Always`, as we use floating tags for our release lines.
+/// This means the tag 23.4 starts of pointing to the same image 23.4.0 does, but switches to 23.4.1 after the releases of 23.4.1.
 pub enum PullPolicy {
-    #[default]
     IfNotPresent,
+    #[default]
     Always,
     Never,
 }
 
 impl ProductImage {
-    pub fn resolve(&self, image_base_name: &str) -> ResolvedProductImage {
+    /// `image_base_name` should be base of the image name in the container image registry, e.g. `trino`.
+    /// `operator_version` needs to be the full operator version and a valid semver string.
+    /// Accepted values are `23.7.0` or `0.0.0-dev`. Versions with a trailing `-pr` or something else are not supported.
+    pub fn resolve(
+        &self,
+        image_base_name: &str,
+        operator_version: &str,
+    ) -> OperatorResult<ResolvedProductImage> {
         let image_pull_policy = self.pull_policy.as_ref().to_string();
         let pull_secrets = self.pull_secrets.clone();
 
         match &self.image_selection {
-            ProductImageSelection::Custom(custom) => {
-                let custom_image_tag = custom
+            ProductImageSelection::Custom(image_selection) => {
+                let image_tag = image_selection
                     .custom
                     .split_once(':')
                     .map_or("latest", |splits| splits.1);
-                let app_version_label = format!("{}-{}", custom.product_version, custom_image_tag);
-                ResolvedProductImage {
-                    product_version: custom.product_version.to_string(),
+                let app_version_label =
+                    format!("{}-{}", image_selection.product_version, image_tag);
+                Ok(ResolvedProductImage {
+                    product_version: image_selection.product_version.to_string(),
                     app_version_label,
-                    image: custom.custom.to_string(),
+                    image: image_selection.custom.clone(),
                     image_pull_policy,
                     pull_secrets,
-                }
+                })
             }
-            ProductImageSelection::StackableVersion(stackable_version) => {
-                let repo = stackable_version
+            ProductImageSelection::StackableVersion(image_selection) => {
+                let repo = image_selection
                     .repo
                     .as_deref()
                     .unwrap_or(STACKABLE_DOCKER_REPO);
+                let stackable_version = image_selection
+                    .stackable_version
+                    .clone()
+                    .unwrap_or(operator_version.to_string());
```

Comment:
```suggestion
                let stackable_version = image_selection
                    .stackable_version
                    .as_deref()
                    .unwrap_or(operator_version);
```

Reply:
I changed this like 3 times and switched back and forth xD Thx!